### PR TITLE
Enable all pyupgrade checks in ruff

### PR DESCRIPTION
**Pull request by tristan957**
This will help to keep us from using deprecated Python features going forward.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -97,5 +97,8 @@ select = [
     "I", # isort
     "W", # pycodestyle
     "B", # bugbear
-    "UP032", # f-string
+    "UP", # pyupgrade
 ]
+
+[tool.ruff.lint.pyupgrade]
+keep-runtime-typing = true # Remove this stanza when we require Python 3.10
```
Comment (tristan957) [inline pyproject.toml:104]: Without this, you get errors suggesting that you use `x | y` instead of `Union[x, y]`

**Review by bayandin — APPROVED** [review]
Nice!

**Comment by tristan957** [comment]
I don't understand why all the test jobs are skipped.

**Comment by tristan957** [comment]
Now it seems fixed.

**Pull request merged**